fem/fe/fe_l2.cpp

```diff
@@ -104,6 +107,7 @@ void L2_QuadrilateralElement::CalcShape(const IntegrationPoint &ip,
    Vector shape_x(p+1), shape_y(p+1);
 #endif
 
+   basis1d.ScaleIntegrated(map_type == VALUE);
```

Comment:
Shouldn't these calls be somewhere in the constructors? Doesn't seem related to the `CalcShape` arguments.

Also, since it's relevant only for `Integrated` bases, I would put it inside `if (basis is integrated)` for better intuition.

Reply:
The issue is that the `basis1d` objects are cached and shared between different finite element objects, so if it's set in the constructor of one finite element, it will actually have an effect on the evaluation of all other finite elements sharing the same basis type and degree.

Reply:
Got it, but I still recommend using an `if` to improve clarity. It might be good also to comment somewhere that the `Integrated` basis is relevant only for `L2` FE (I assume that's the case since the calls are not made in the `H1_` FE classes).